code/game/objects/items/devices/PDA/PDA_types.dm

```diff
@@ -6,13 +6,26 @@
 	icon_state = "pda-clown"
 	desc = "A portable microcomputer by Thinktronic Systems, LTD. The surface is coated with polytetrafluoroethylene and banana drippings."
 	ttone = "honk"
+	var/slipvictims = list() //Haven - Ports crew objectives from cit. Credit to Deathride58
 
 /obj/item/device/pda/clown/Initialize()
 	. = ..()
-	AddComponent(/datum/component/slippery, 120, NO_SLIP_WHEN_WALKING, CALLBACK(src, .proc/AfterSlip))
 
-/obj/item/device/pda/clown/proc/AfterSlip(mob/living/carbon/human/M)
-	if (istype(M) && (M.real_name != owner))
+	AddComponent(/datum/component/slippery, 120, NO_SLIP_WHEN_WALKING)
+
+/obj/item/device/pda/clown/ComponentActivated(datum/component/C)
+	..()
+	var/datum/component/slippery/S = C
+	if(!istype(S))
+		return
+	var/mob/living/carbon/human/M = S.slip_victim
+	if (istype(M) && (M.real_name != src.owner))
+		slipvictims |= M	//Haven - Ports crew objectives from cit. Credit to Deathride58
```

Comment:
This line right here is the only one that's necessary to track how many unique people have been slipped.

Reply:
Nnoot parsing what needs fixing here.